services/app/src/components/team/TeamMember.jsx

```diff
@@ -24,7 +30,8 @@ const TeamMember = ({ name, id, skipMove, color, status, removePlayer, index })
       </div>
     </div>
     {name}
-    { status === playerStatuses.FigurePlaced && <img src={CheckMark} className="team-list-item-icon" alt="player done icon" /> }
+    { (status === playerStatuses.FigurePlaced && !isAllTurnsMade) && <div className="team-list-item-icon" alt="player done icon"> <BouncingDots/></div>}
```

Comment:
Lets introduce a new component for player status and move all the logic related to user status display into that component.